### PR TITLE
Fix crate_universe's `all_crate_deps` and `aliases` functions failing in case the crate's Cargo.toml has condtional dependencies

### PR DESCRIPTION
**Pull request by k1nkreet**
In this PR I've added: 

  * Example of the issue with cargo_universe used for a crate with conditional dependencies described in #1880 into the `examples/`
  * Fix for `Error: unhashable type: 'list'` developed in #1901 
  * Fix for invalid platform labels generated for _CONDITIONS dictionary mentioned [here](https://github.com/bazelbuild/rules_rust/pull/1901#issuecomment-1492029546)

### CODE REVIEW

File: crate_universe/src/rendering/templates/module_bzl.j2 [hunk 1]
```diff
@@ -313,7 +314,7 @@ _BUILD_PROC_MACRO_DEPENDENCIES = {% set deps_type = "build-proc-macro" %}{% incl
 _BUILD_PROC_MACRO_ALIASES = {% set deps_type = "build-proc-macro" %}{% include "partials/module/aliases_map.j2" %}
 
 _CONDITIONS = {
-{%- for condition, triples in context.conditions %}
+{%- for condition, triples in platforms %}
```
Comment (UebelAndre) [inline crate_universe/src/rendering/templates/module_bzl.j2:317]: Can you update the `crates_vendor` examples so I can see a rendering of this?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thanks for putting up the pull-request to revive the effort!

File: crate_universe/src/rendering/templates/module_bzl.j2 (hunk 1 above)
Comment (k1nkreet) [reply crate_universe/src/rendering/templates/module_bzl.j2:320]: Done

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Looks good to me, thank you so much!

**Review by UebelAndre — APPROVED** [review]

**Pull request merged**